wirelesshid.js

```diff
@@ -120,12 +124,31 @@ var HID = GObject.registerClass({
         return new Clutter.ColorizeEffect({tint: color});
     }
 
-    _update() {
+    _checkBatteryPresent() {
+        let isBatteryPresent = true;
+        if (this.device.is_present === false) {
+            isBatteryPresent = false;
+        }
+
+        //Some devices report 'present' as true, even if no battery is present
+        //Instead, check the state and icon name to try and find these
+        if (this.device.state === UPower.DeviceState.UNKNOWN && this.device.iconName === "battery-missing-symbolic") {
```

Comment:
My Logitech Mouse MX and Logitech MX Keys both have this Unknown state and also the missing icon...

Like this:
```
  keyboard
    present:             yes
    rechargeable:        no
    state:               unknown
    warning-level:       none
    percentage:          100%
    icon-name:          'battery-missing-symbolic'

  mouse
    present:             yes
    rechargeable:        no
    state:               unknown
    warning-level:       none
    percentage:          65%
    icon-name:          'battery-missing-symbolic'


```
Those  two devices are the reason why I created this extension:-) So this condition is definitely not wanted. Obviously, we can not rely on the attributes. I am afraid removing the condition will break your feature, right?

Reply:
It would break the detection of receivers with removable devices, yes.

Reply:
Should be sorted by the latest commit